wallet/wallet.go

```diff
@@ -2989,6 +2989,23 @@ func isRevocation(tx *wire.MsgTx) bool {
 	return b
 }
 
+// hasVotingAuthority returns whether the 0th output of a ticket purchase can be
+// spent by a vote or revocation created by this wallet.
+func (w *Wallet) hasVotingAuthority(addrmgrNs walletdb.ReadBucket, ticketPurchase *wire.MsgTx) (bool, error) {
+	out := ticketPurchase.TxOut[0]
+	_, addrs, _, err := txscript.ExtractPkScriptAddrs(out.Version,
+		out.PkScript, w.chainParams)
+	if err != nil {
+		return false, err
+	}
+	for _, a := range addrs {
+		if w.Manager.ExistsHash160(addrmgrNs, a.Hash160()[:]) {
```

Comment:
if P2SH, we're assuming it's 1-of-N? not that there is any other kind of multisig addr ticket that makes any sense I guess